spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskDeleteService.java

```diff
@@ -270,13 +277,21 @@ public void deleteTaskExecutions(Set<Long> taskExecutionIds) {
 
 		// Delete Task Related Data
 
-		auditData.put("Deleted # of Task Executions", taskExecutionIdsWithChildren.size());
-		auditData.put("Deleted Task Execution IDs", StringUtils.collectionToDelimitedString(taskExecutionIdsWithChildren, ", "));
 		final int numberOfDeletedTaskExecutionParamRows = dataflowTaskExecutionDao.deleteTaskExecutionParamsByTaskExecutionIds(taskExecutionIdsWithChildren);
 		final int numberOfDeletedTaskTaskBatchRelationshipRows = dataflowTaskExecutionDao.deleteTaskTaskBatchRelationshipsByTaskExecutionIds(taskExecutionIdsWithChildren);
-		final int numberOfDeletedTaskManifestRows = this.dataflowTaskExecutionMetadataDao.deleteManifestsByTaskExecutionIds(taskExecutionIdsWithChildren);
+
+		int numberOfDeletedTaskManifestRows = 0;
+		for (Long taskExecutionId : taskExecutionIdsWithChildren) {
+			numberOfDeletedTaskManifestRows += this.taskManifestRepository.deleteTaskExecutionManifestByTaskExecutionId(taskExecutionId);
+		}
+
+		this.entityManager.flush();
```

Comment:
Could you elaborate on why this is necessary?

Reply:
Delete below done via `dataflowTaskExecutionDao` would not even be in a same transaction, afaik, as `JdbcTemplate` and hibernate are probably working via different jdbc connections. I don't think you can do delete operations like this.

Reply:
The `JpaTransactionManager` (which SCDF is using) supports the same transaction across the two.  You can read more about it in the documentation for it here: https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/orm/jpa/JpaTransactionManager.html

Reply:
Apart from `TaskExecutionsDocumentation.taskExecutionRemoveAndTaskDataRemove()` which seem to fail all the time when I commented out that flush there's no test failures. It might work better if you try to delete the actual entity instead of trying to create a query for it. There has to be better way that hacking into EntityManager.

Reply:
In a skipper's `Release` entity we used these in a field referencing to other table:
```
@OneToOne(cascade = { CascadeType.ALL })
@JoinColumn(foreignKey = @ForeignKey(name = "fk_release_manifest"))
private Manifest manifest;
````
Not sure if it helps but we need something to tell hibernate that it should not preoptimize db calls.

Reply:
I'm not sure I follow.  What do you mean by "delete the actual entity".  That's what I'm doing.  Unless you're proposing I do a find for the entity, then a delete but that will result in the same issue.  For the skipper example you are sharing, those are all entitites, correct?  Which would mean that JPA would handle it all.  The `TaskExecution` that the `TaskExecutionManifest` is not a JPA entity which is why this issue arises in the first place.

Reply:
With `deleteTaskExecutionManifestByTaskExecutionId` you're essentially creating a query which work differently if you get access to a full entity object and then try to delete by passing that into `taskExecutionManifestRepository` delete method. Just a theory when things might work better. It'd difficult to try different ways to make this work better as no tests fail if that `flush` is removed. We should not have these kind of workaround in place which we cannot test.

Reply:
Looks like I was able to work without explicit flush by adding explicit new propagation to `TaskExecutionManifestRepository`:

```
@Transactional(propagation = Propagation.REQUIRES_NEW)
int deleteTaskExecutionManifestByTaskExecutionId(long taskExecutionid);
```

See if that work for you as well!

Reply:
There were some notes at https://docs.jboss.org/hibernate/core/3.3/reference/en/html/objectstate.html#objectstate-flushing so had an idea that if we wrap this in a new transaction hibernate then sees that boundary and flushes. It's still a hack but we'd be working with a system, not around it.

Reply:
1.  I can add a test to validate the transaction rollback for the method (We don't do this anywhere else that I'm aware of and this PR should be functionally the same as the code before which is why I didn't add it).
2. Changing the transaction propagation would be incorrect since the entire service method needs to be in a single transaction, committed or rolled back as one.  The current transaction as configured should work that way as coded.  Changing the propagation would prevent that I think.
3. I don't see how adding an explicit call to flush is working around it.  That documentation you point out explicitly states that a call to flush is a way to make this work.

All of this being said, this is one line of code.  I brought this up in the slack channel before adding it and @markpollack seemed ok with it at the time.  Any of these other methods seem like overkill for such a simple thing IMHO.